cmd/build.go

```diff
@@ -303,6 +303,13 @@ func (c buildConfig) Prompt() (buildConfig, error) {
 	// value and will always use the value from the config (flag or env variable).
 	// This is not strictly correct and will be fixed when Global Config: Function
 	// Context is available (PR#1416)
+	hasFunc, err := fn.IsFunctionInitialized(c.Path)
+	if err != nil {
+		return c, err
+	}
+	if !hasFunc {
+		return c, fmt.Errorf("no function has been initialized in the current directory. Please initialize a function by running either:\n- func init --language <your language>\n- func create <function name> --language <your_language>")
+	}
 	f, err := fn.NewFunction(c.Path)
```

Comment:
Should not this already return `ENOENT`?

Reply:
I think we could simply check `f.Initialized()` instead of introducing new `IsFunctionInitialized()` function.

Reply:
TBH I am surprised that the function returns `err==nil` when there is no function. Why it does that @lkingland ? https://github.com/knative/func/blob/e6fa020f7833e62c29364902da04e7cc6f7dab60/pkg/functions/function.go#L257-L265

Reply:
@lkingland also if you mean zero-value by "default function" , then you also need to zero the path of the f.

Reply:
Yes, good question.  A Function is first a structure in memory, which can be serialized to disk.  The path to where the Function is, or will be when initialized, is the only value provided in the constructor, and thus is the only field that is populated with user data when creating a new function from scratch.


Like you say.  Just check if `f.Initialized()`.

Reply:
I still do not like, but can live with it. Just the function docstring says:
>NewFunction from a given path. Invalid paths, or no function at path are errors.

Reply:
I bet that if you asked 10 programmers what `NewFunction(pathWithNoFunction)`  returns at least 9 would tell it's an error.